backend/utils/natural_language_processing_api.py

```diff
@@ -0,0 +1,44 @@
+from dotenv import load_dotenv
+import json
+from groq import Groq, AuthenticationError, APIConnectionError
+import os
+
+
+def prompt_nlp_model(prompt, api_inputted_key): 
+    """
+    Prompts the API and with information
+
+    Args:
+        str: A question or command for the model to fulfill
+
+    Returns:
+        JSON: a dictionary with the information regarding the prompt and response
+    """
+
+    #create an instance of groq with our API key
+    client = Groq(api_key= api_inputted_key,)
+    
+    try:
+        #API call
+        api_prompt = client.chat.completions.create(
+            messages=[
+                {
+                    "role": "user",
+                    "content": prompt,
+                }
+            ],
+            model="llama3-8b-8192",
+
+        )
+
+    #if the API key is invalid
+    except AuthenticationError:
+        return "API KEY IS INVALID"
```

Comment:
I wouldn't return a string as the error.
i would return `{"error": "API KEY is invalid"}`